### PR TITLE
Feature/ab#114 save relation on target entity

### PR DESCRIPTION
**Pull request by witchpou**
# Save relation bidirectional, Delete relations

## Description

* backend was adopted to save relation on target entity
* frontend was changed to delete relations

## How has this been tested?
Manually

## Screenshots (if appropriate):
image

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.

### CODE REVIEW

File: webclient/app/src/app/commons/entityDialog/EntityDialog.jsx [hunk 1]
```diff
@@ -139,8 +145,12 @@ function EntityDialog(props) {
...
             .then(() => {
-                onClose()
-                setIsSaving(false);
+                entityRest.findAllEntitiesByApp(appId)
+                    .then((response) => {
+                        handleUpdateEntities(response.data);
+                        onClose();
+                        setIsSaving(false);
+                    })
             })
```
Comment (DerCed) [inline webclient/app/src/app/commons/entityDialog/EntityDialog.jsx:154]:
```suggestion
                return entityRest.findAllEntitiesByApp(appId)
            })
            .then((response) => {
                        handleUpdateEntities(response.data);
                        onClose();
                        setIsSaving(false);
                    })
```

File: webclient/app/src/app/commons/entityDialog/EntityDialog.jsx [hunk 2]
```diff
@@ -240,7 +257,7 @@ function EntityDialog(props) {
         if (!entity.fields || entity.fields.length <= 0) {
             return (
                 <div className={entityEditorStyles.statementWrapper}>
-                    <Statement icon={<CheckBoxOutlineBlank/>} message={t("entityDialog.noFields")}/>
+                    <Statement message={t("entityDialog.noFields")}/>
```
Comment (DerCed) [inline webclient/app/src/app/commons/entityDialog/EntityDialog.jsx:260]: An icon is necessary here. Maybe a Plus icon or something else would be great here. https://mui.com/components/material-icons/
```suggestion
                    <Statement icon={<Add/>} message={t("entityDialog.noFields")}/>
```

File: webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx [hunk 3]
```diff
@@ -39,93 +41,105 @@ function RelationshipAccordion(props) {
...
-                                onChange={(event) => editRelationshipProperty("otherEntityRelationshipName", event.target.value)}
-                                helperText={ t("relationship.targetField.description", {entityName: currentEntity.name ? currentEntity.name : t("relationship.sourceEntity")} )}
-                                regex={RegexConfig.relationship}
-                            />
-                        </FormControl>
-                    </Grid>
-                </Grid>
-            </AccordionDetails>
-        </Accordion>
+        <Grid container>
```
Comment (DerCed) [inline webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx:44]:
```suggestion
        <Grid container spacing={5}>
```

File: webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx [hunk 4]
```diff
@@ -39,93 +41,105 @@ function RelationshipAccordion(props) {
...
+            <Grid item sm={11}>
```
Comment (DerCed) [inline webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx:45]:
```suggestion
            <Grid item xs={11}>
```

File: webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx [hunk 5]
```diff
@@ -39,93 +41,105 @@ function RelationshipAccordion(props) {
...
+                                        helperText={ t("relationship.targetField.description", {entityName: currentEntity.name ? currentEntity.name : t("relationship.sourceEntity")} )}
+                                        regex={RegexConfig.relationship}
+                                    />
+                                </FormControl>
+                            </Grid>
+                        </Grid>
+                    </AccordionDetails>        
+                </Accordion>
+            </Grid>
+            <Grid item sm={1}>
```
Comment (DerCed) [inline webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx:134]:
```suggestion
            <Grid item xs={1}>
```

File: webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx [hunk 6]
```diff
@@ -39,93 +41,105 @@ function RelationshipAccordion(props) {
...
+                <IconButton
+                    aria-label="delete"
+                    onClick={handleDelete}
+                >
+                    <Delete />
```
Comment (DerCed) [inline webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx:139]: Im not quite happy with the current structuring of the delete button. It is certainly good for now but maybe we have to implement this button in the Accordion itself due to Responsive design issues.

File: webclient/app/src/app/commons/statement/Statement.jsx [hunk 7]
```diff
@@ -19,7 +19,7 @@ function Statement(props) {
         <div className={statementStyles.root}>
             <div className={statementStyles.content}>
                 {React.cloneElement(icon, {fontSize: "large"})}
-                <Typography className={statementStyles.message} gutterBottom>{message}</Typography>
+                <Typography component={'span'} className={statementStyles.message} gutterBottom>{message}</Typography>
```
Comment (DerCed) [inline webclient/app/src/app/commons/statement/Statement.jsx:22]: Why did you add span here?

File: webclient/app/src/app/pages/appEditor/AppEditor.jsx [hunk 8]
```diff
@@ -103,8 +103,9 @@ function AppEditor() {
                     appId={+appId}
                     entities={entities}
                     coordinates={entityRelationCoordinates}
-                    updateCoordinates={(entities) => updateEntityRelationCoordinates(entities)}
-                    handleUpdateEntities={updatedEntities => setEntities(updatedEntities)}
+                    handleUpdateEntities={updatedEntities => {
+                        handleUpdateEntities(updatedEntities);
+                    }}
```
Comment (DerCed) [inline webclient/app/src/app/pages/appEditor/AppEditor.jsx:108]:
```suggestion
                    handleUpdateEntities={handleUpdateEntities}
```

File: webclient/app/src/app/pages/appEditor/AppEditor.jsx [hunk 9]
```diff
@@ -115,10 +116,9 @@ function AppEditor() {
                 <ConclusionSection
                     appId={+appId}
                     entities={entities}
+                    coordinates={updateRelationCoordinates(entities)}
```
Comment (DerCed) [inline webclient/app/src/app/pages/appEditor/AppEditor.jsx:119]: ``coordinates`` does not imply, that this should handle an update

File: webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx [hunk 10]
```diff
@@ -63,22 +46,11 @@ function ErDesigner(props) {
...
-            })
-        }
-
         return entityRest.delete(entityId)
-            .then((response) => {
-                return entityRest.findAllEntitiesByApp(appId)
-            })
-            .then(response => {
-                handleUpdateEntities(response.data)
+            .then(() => {
```
Comment (DerCed) [inline webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx:50]: Why did you change that? This is exactly what it did before

File: webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx [hunk 11]
```diff
@@ -94,17 +66,9 @@ function ErDesigner(props) {
             delete updatedEntity.id;
             updatedEntity.isNewEntity = false;
         }
-        setCurrentEntity(updatedEntity)
-
-        return entityRest.createEntityByApp(appId, updatedEntity)
```
Comment (DerCed) [inline webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx:99]: Why did you remove the update process?

File: webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx [hunk 12]
```diff
@@ -115,23 +79,14 @@ function ErDesigner(props) {
...
-
-        updateCoordinates(entities);
-    }
-
-    function renderRelations() {
-
-        return coordinates.map((coordinate, index) => {
-            return <SteppedLineTo from={coordinate.from} to={coordinate.to} borderColor={theme.palette.primary.main} borderWidth={theme.palette.line.width} />
-        })
+        entityRest.updateEntityByAppId(appId, entity);
```
Comment (DerCed) [inline webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx:82]: The updated data from the server is not injected into the state. Maybe a update of the state from the server would be great here

File: webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx [hunk 13]
```diff
@@ -115,23 +79,14 @@ function ErDesigner(props) {
...
 
+        handleUpdateEntities(entities);
     }
 
-
     function renderEntities() {
         if (entities.length === 0) {
-            return <Statement message={"No entities found"} icon={<CheckBoxOutlineBlank/>}/>
+            return <Statement message={"No entities found"}/>
```
Comment (DerCed) [inline webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx:89]:
```suggestion
            return <Statement icon={<Plus/>} message={"No entities found"}/>
```

File: webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx [hunk 14]
```diff
@@ -221,13 +178,16 @@ function ErDesigner(props) {
...
             {renderEntities()}
-            {renderRelations()}
+            {renderRelations(coordinates, theme)}
         </div>
         <EntityDialog
             entityId={currentEntity?.id}
             onClose={closeEntityDialog}
             handleSave={(data) => updateEntity(data)}
             entities={entities}
+            handleUpdateEntities={(updatedEntities) => handleUpdateEntities(updatedEntities)}
```
Comment (DerCed) [inline webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx:188]:
```suggestion
            handleUpdateEntities={handleUpdateEntities}
```

File: webclient/app/src/localization/translations-de-DE.js [hunk 15]
```diff
@@ -68,12 +68,12 @@ const translationsDeDE = {
     "relationship.relationType": "Relationstyp",
     "relationship.sourceEntity": "Entität - Quelle der Relation",
     "relationship.sourceField": "Relationsname",
-    "relationship.sourceField.description": "Name des Feldes vom Typ {{entityName}}",
+    "relationship.sourceField.description": "Name des Feldes vom Typ {{entityName}}. Groß und Kleinschreibung mit Zahlen erlaubt. Erstes Zeichen muss ein großer Buchstabe sein.",
```
Comment (DerCed) [inline webclient/app/src/localization/translations-de-DE.js:71]: Translations are not applied to english language

File: webclient/app/src/app/pages/appEditor/HandleRelations.jsx [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+                coordinates.push({
+                    from: 'anchor_' + entity.name, to: 'anchor_' + relationship.otherEntityName
+                })
+            })
+        }
+    })
+    return coordinates;
+}
+
+function renderRelations(coordinates, theme) {
```
Comment (DerCed) [inline webclient/app/src/app/pages/appEditor/HandleRelations.jsx:17]: Why did you export this to an extra file? This is just used inside the ERSection

File: webclient/app/src/app/pages/appEditor/sections/conclusion/ConclusionSection.jsx [hunk 17]
```diff
@@ -73,9 +73,10 @@ function ConclusionSection(props) {
             </Grid>
             <Typography variant={"h4"} gutterBottom>{t("app.erdiagram")}</Typography>
             <ErSection entities={entities} 
+                appId={+appId}
                 coordinates={coordinates}
-                updateCoordinates={updateCoordinates} 
-                dense editable={false}/>
+                dense editable={false}
```
Comment (DerCed) [inline webclient/app/src/app/pages/appEditor/sections/conclusion/ConclusionSection.jsx:78]:
```suggestion
                dense 
                editable={false}
```

**Review by DerCed — CHANGES_REQUESTED** [review]
I think that this Pull Request is way too large and covers multiple tickets at once or even unspecified ones. I would ask you to make smaller tickets so that it is easier to spot errors. There are also a couple of errors I haven't spotted the last time e.g. in Relationship.js

File: webclient/app/src/app/commons/entityDialog/EntityDialog.jsx (hunk 1 above)
Comment (witchpou) [reply webclient/app/src/app/commons/entityDialog/EntityDialog.jsx:154]: Why I should return a promise to an onClick call?

File: webclient/app/src/app/commons/entityDialog/EntityDialog.jsx (hunk 2 above)
Comment (witchpou) [reply webclient/app/src/app/commons/entityDialog/EntityDialog.jsx:260]: there is a default icon saved in the component
